File: android/app/src/main/java/fr/gshz/hideandseek/domain/model/GameSize.kt
package fr.gshz.hideandseek.domain.model

enum class GameSize(val wireValue: String) {
    Small("S"),
    Medium("M"),
    Large("L"),
    ;

    companion object {
        fun fromWireValue(value: String): GameSize = entries.first { it.wireValue == value }
    }
}

File: backend/src/Enum/GameSize.php
<?php

declare(strict_types=1);

namespace App\Enum;

enum GameSize: string
{
    case Small = 'S';
    case Medium = 'M';
    case Large = 'L';

    public function ordinal(): int
    {
        return match ($this) {
            self::Small => 0,
            self::Medium => 1,
            self::Large => 2,
        };
    }
}
